Tests/LibAudio/TestWav.cpp

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright (c) 2024, Lee Hanken <github-12-2017-ds8@leehanken.uk>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <AK/ByteString.h>
+#include <AK/LexicalPath.h>
+#include <AK/ScopeGuard.h>
+#include <LibAudio/WavLoader.h>
+#include <LibAudio/WavWriter.h>
+#include <LibCore/Directory.h>
+#include <LibCore/File.h>
+#include <LibFileSystem/FileSystem.h>
+#include <LibFileSystem/TempFile.h>
+#include <LibTest/TestCase.h>
+
+static void read_wave(int const num_samples, u32 const rate, int channels, ByteString const in_path, ByteString const out_path = "")
+{
+
+    auto loader = TRY_OR_FAIL(Audio::Loader::create(in_path));
+
+    Optional<AK::NonnullOwnPtr<Audio::WavWriter>> maybe_writer;
+
+    if (!out_path.is_empty()) {
+        maybe_writer = TRY_OR_FAIL(Audio::WavWriter::create_from_file(out_path, rate, channels));
+    }
+
+    int samples_read = 0;
+    int size = 0;
+
+    do {
+        auto maybe_samples = loader->get_more_samples();
+        auto samples = maybe_samples.release_value();
+
+        if (maybe_writer.has_value()) {
+            TRY_OR_FAIL(maybe_writer.value()->write_samples(samples.span()));
+        }
+
+        size = samples.size();
+        samples_read += size;
+
+    } while (size);
+
+    if (maybe_writer.has_value()) {
+        TRY_OR_FAIL(maybe_writer.value()->finalize());
+    }
+
+    EXPECT_EQ(samples_read, num_samples);
+}
+
+static void compare_files(ByteString const in_path, ByteString const out_path)
+{
+    Array<u8, 4096> buffer1;
+    Array<u8, 4096> buffer2;
+
+    auto original_file = MUST(Core::File::open(in_path, Core::File::OpenMode::Read));
+    auto copied_file = MUST(Core::File::open(out_path, Core::File::OpenMode::Read));
+
+    while (!original_file->is_eof() && !copied_file->is_eof()) {
+        auto original_bytes = TRY_OR_FAIL(original_file->read_some(buffer1));
+        auto copied_bytes = TRY_OR_FAIL(copied_file->read_some(buffer2));
+
+        EXPECT_EQ(original_bytes, copied_bytes);
+    }
+
+    original_file->close();
```

Comment:
If we are keeping the `File` reading part, this is not needed: `File` are closed when the destructor is called.